torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -694,6 +698,26 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
 
   void release_resources() override;
 
+  // ProcessGroups optionally can be "bound" to a specific device.
+  // Currently this is only for nccl and allows for some opt-in
+  // optimizations such as automatic use of ncclCommSplit.  The device
+  // is specified in `init_process_group` and eventually makes it
+  // here and then down into the actual backend instances.
+  c10::optional<at::Device> getBoundDeviceId() const {
+    return bound_device_id_;
+  }
+
+  void setBoundDeviceId(c10::optional<at::Device> device) {
+    if (device) {
+      TORCH_CHECK(device->has_index(), "setBoundDeviceId must have an index");
+    }
+    bound_device_id_ = device;
+  }
+
+  bool supportsSplitting() const {
+    return backendType_ == c10d::ProcessGroup::BackendType::NCCL;
+  }
```

Comment:
See other comment that "Ideally support_splitting should tie to backend."
If we made that change, then we can let each backend to return true or false (than manually deciding here).